internal/controllers/ipaddressclaim.go

```diff
@@ -211,57 +211,60 @@ func (r *IPAddressClaimReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		}
 	}()
 
-	var pool pooltypes.GenericInClusterPool
+	if !controllerutil.ContainsFinalizer(claim, ReleaseAddressFinalizer) {
+		controllerutil.AddFinalizer(claim, ReleaseAddressFinalizer)
+	}
 
-	if claim.Spec.PoolRef.Kind == inClusterIPPoolKind {
-		icippool := &v1alpha2.InClusterIPPool{}
-		if err := r.Client.Get(ctx, types.NamespacedName{Namespace: claim.Namespace, Name: claim.Spec.PoolRef.Name}, icippool); err != nil && !apierrors.IsNotFound(err) {
-			return ctrl.Result{}, errors.Wrap(err, "failed to fetch pool")
-		}
-		pool = icippool
-	} else if claim.Spec.PoolRef.Kind == globalInClusterIPPoolKind {
-		gicippool := &v1alpha2.GlobalInClusterIPPool{}
-		if err := r.Client.Get(ctx, types.NamespacedName{Name: claim.Spec.PoolRef.Name}, gicippool); err != nil && !apierrors.IsNotFound(err) {
-			return ctrl.Result{}, errors.Wrap(err, "failed to fetch pool")
+	pool, err := r.findPool(ctx, claim)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err := errors.New("pool not found")
+			log.Error(err, "the referenced pool could not be found")
+			if !claim.ObjectMeta.DeletionTimestamp.IsZero() {
+				return r.reconcileDelete(ctx, claim)
+			}
+			return ctrl.Result{}, nil
 		}
-		pool = gicippool
+		return ctrl.Result{}, errors.Wrap(err, "failed to fetch pool")
 	}
 
 	if pool != nil && annotations.HasPaused(pool) {
 		log.Info("IPAddressClaim references Pool which is paused, skipping reconciliation.", "IPAddressClaim", claim.GetName(), "Pool", pool.GetName())
 		return ctrl.Result{}, nil
 	}
 
-	address := &ipamv1.IPAddress{}
-	if err := r.Client.Get(ctx, req.NamespacedName, address); err != nil && !apierrors.IsNotFound(err) {
-		return ctrl.Result{}, errors.Wrap(err, "failed to fetch address")
+	if !claim.ObjectMeta.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, claim)
 	}
 
-	if !controllerutil.ContainsFinalizer(claim, ReleaseAddressFinalizer) {
-		controllerutil.AddFinalizer(claim, ReleaseAddressFinalizer)
-	}
+	return r.reconcileNormal(ctx, claim, pool)
+}
 
-	if !claim.ObjectMeta.DeletionTimestamp.IsZero() {
-		return r.reconcileDelete(ctx, claim, address)
+func (r *IPAddressClaimReconciler) findPool(ctx context.Context, claim *ipamv1.IPAddressClaim) (pooltypes.GenericInClusterPool, error) {
+	if claim.Spec.PoolRef.Kind == inClusterIPPoolKind {
+		icippool := &v1alpha2.InClusterIPPool{}
+		if err := r.Client.Get(ctx, types.NamespacedName{Namespace: claim.Namespace, Name: claim.Spec.PoolRef.Name}, icippool); err != nil {
+			return nil, err
+		}
+		return icippool, nil
+	} else if claim.Spec.PoolRef.Kind == globalInClusterIPPoolKind {
+		gicippool := &v1alpha2.GlobalInClusterIPPool{}
+		if err := r.Client.Get(ctx, types.NamespacedName{Name: claim.Spec.PoolRef.Name}, gicippool); err != nil {
+			return nil, err
+		}
+		return gicippool, nil
 	}
+	return nil, fmt.Errorf("Unknown pool type: %s", claim.Spec.PoolRef.Kind)
```

Comment:
Error strings should be lowercase (the linter complains).

Reply:
Fixed! Thanks!